muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/guider/body/application/notes.tsx

```diff
@@ -0,0 +1,115 @@
+import * as React from "react";
+import { useDispatch, useSelector } from "react-redux";
+import { useTranslation } from "react-i18next";
+import {
+  loadNotes,
+  updateNote,
+  updateRecipientNoteStatus,
+  toggleNoteArchive,
+} from "~/actions/main-function/guider";
+import { StateType } from "~/reducers";
+import NotesItemList from "~/components/general/notes/notes-item-list";
+import {
+  UpdateNoteRequest,
+  UpdateNoteReceiverRequest,
+} from "~/generated/client";
+import { GuiderContext } from "../../context";
+import { useFilterNotes } from "~/components/guider/hooks/useFilterNotes";
+/**
+ * GuiderStudentsProps
+ */
+interface GuiderNotesProps {}
+
+/**
+ * GuiderStudents
+ * @param props props
+ */
+const GuiderNotes = (props: GuiderNotesProps) => {
```

Comment:
From performance perspective useCallback won't add any effect if NotesItemList is not memoized. This can be done using React.memo().

For same reason, any non primitive types needs to be memoized also. notes is already memoized by custom hook as it seems.

Note, that if this components parent re-renders also, then it won't make difference because all is re-rendered anyways. Even memoizing GuiderNotes doesn't help, as there is not any props to compare. You must evaluate if memoizing is worth to do, so if component tree is deep, calculation are expensive etc...